perses/rjmc/geometry.py

```diff
@@ -660,29 +715,130 @@ def _angle_logq(self, theta, angle, beta):
         beta : float
             1/kT or inverse temperature
         """
-        k_eq = angle.type.k
-        theta0 = angle.type.theteq
-        logq = -beta*k_eq*0.5*(theta-theta0)**2
-        return logq
-
-    def _propose_bond(self, bond, beta):
+        # import scipy.integrate as integrate
+        #
+        # k_eq = angle.type.k.in_units_of(units.kilocalorie_per_mole/units.radians**2) * self._angle_softening_constant
+        # theta0 = angle.type.theteq.in_units_of(units.radians)
+        # theta_target=theta.in_units_of(units.radians)
+        #
+        #
+        # array=np.linspace(-np.pi, np.pi)
+        # #exponand=np.exp(-1*beta*k_eq*0.5*(array-theta0)**2)
+        #
+        # f=lambda x: np.exp(-beta*k_eq*0.5*(x*units.radians-theta0)**2)*np.sin(x)
+        #
+        # integral, integral_err=integrate.quad(f,-np.pi,np.pi)
+        # logZ_theta=np.log(integral)
+        #
+        #
+        # logq = np.log(np.sin(theta_target/units.radians))-beta*0.5*k_eq*(theta_target-theta0)**2-logZ_theta
+        # return logq
+
+        theta0 = angle.type.theteq.value_in_unit_system(units.md_unit_system);
+        #print('theta0: ', theta0)
+        k = angle.type.k.value_in_unit_system(units.md_unit_system) * self._angle_softening_constant
+        sigma_theta = units.sqrt(1.0/(beta*k).value_in_unit_system(units.md_unit_system))
+        #print('sigma_theta', sigma_theta)
+        #theta = sigma_theta*np.random.randn() + theta0
+        lower_bound, upper_bound=0., np.pi
+
+
+        #'exact' probability
+        f=lambda x: np.sin(theta)*np.exp(-(0.5/sigma_theta**2)*(x-theta0)**2)
+        division_size=np.pi/n_divisions
+        theta_array=np.linspace(0,np.pi, n_divisions)
+        theta_index=np.argmin(np.abs(theta-theta_array))
+        Z=sum(f(theta_array))
+        #integral, integral_err=integrate.quad(f,lower_bound,upper_bound)
+        #logZ_theta=np.log(integral)
+        #print('theta: ', theta)
+
+        logp = np.log(np.sin(theta_array[theta_index])*np.exp(-(0.5/sigma_theta**2)*(theta_array[theta_index]-theta0)**2))-np.log(division_size)-np.log(Z)
```

Comment:
This can be 
```python
logp = np.log(np.sin(theta_array[theta_index])) - (0.5/sigma_theta**2)*(theta_array[theta_index]-theta0)**2) - np.log(division_size) - np.log(Z)
```